plugins/woocommerce-blocks/tests/e2e/tests/checkout/order-confirmation.block_theme.side_effects.spec.ts

```diff
@@ -109,9 +114,19 @@ test.describe( 'Shopper → Order Confirmation (logged in user)', () => {
 		// Confirm order details are not visible
 		await pageObject.verifyOrderConfirmationDetails( page, false );
 
-		// The following tests are skipped until the multiple sign in roles is implemented
-		// - Confirm details are hidden when logged out
-		// - Confirm data is hidden without valid session/key
+		await test.step( 'Logout the user and revisit the order received page to verify that details are displayed when woocommerce_order_received_verify_known_shoppers is disabled', async () => {
+			await requestUtils.activatePlugin(
+				'woocommerce-blocks-test-order-confirmation-filters'
+			);
+			await page.goto( '/my-account' );
+			await page
+				.locator(
+					'li.woocommerce-MyAccount-navigation-link--customer-logout a'
+				)
+				.click();
+			await page.goto( orderReceivedURL );
```

Comment:
Let's ensure we land on the logged-out page before going to `orderReceivedURL`. This should prevent the test from behaving flaky by ensuring we're fully logged out before continuing. We can use [`page.waitForURL`](https://playwright.dev/docs/api/class-frame#frame-wait-for-url) to for that:

```suggestion
			await page
				.locator(
					'li.woocommerce-MyAccount-navigation-link--customer-logout a'
				)
				.click();
			await page.waitForURL( /* PUT THE EXPECTED PATH HERE */ );
			await page.goto( orderReceivedURL );
```

Reply:
We can also use the `await expect( page ).toHaveURL( /* logged-out URL */ ),` which will work the same way, only producing a different output in case of an error. Both checks are good, although I'd lean towards using the `expect` assertion as it makes more sense semantically here - we want to be sure the user is logged out.

Reply:
Nice catch, done in https://github.com/woocommerce/woocommerce/pull/46957/commits/a69def91c119e7999a4e87db5ac0a3023d07a9d2 👍🏼